src/erlcloud_aws.erl

```diff
@@ -967,3 +989,149 @@ error_msg( Message ) ->
 error_msg( Format, Values ) ->
     Error = iolist_to_binary( io_lib:format( Format, Values ) ),
     throw( {error, Error} ).
+
+
+
+%%====================================================================
+%% Test Functions
+%%====================================================================
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+
+
+default_profile_test_() ->
+    {setup, fun profiles_test_setup/0, fun profiles_test_cleanup/1,
```

Comment:
You could have used `foreach` type of fixture to share setup/teardown functions among tests cases and avoid repeating the code.